File: ApiTiketero/Ticketero/src/Ticketero.Domain/Entities/ActivoFijo.cs
using System.ComponentModel.DataAnnotations;

namespace Ticketero.Domain.Entities;

public class ActivoFijo : EntityBase
{
    public int KioskoId { get; set; }

    [StringLength(50)]
    public string TipoActivo { get; set; } = string.Empty;

    [StringLength(50)]
    public string NumeroActivo { get; set; } = string.Empty;

    [StringLength(200)]
    public string? Descripcion { get; set; }

    [StringLength(100)]
    public string? Marca { get; set; }

    [StringLength(100)]
    public string? Modelo { get; set; }

    [StringLength(100)]
    public string? Serie { get; set; }

    [StringLength(100)]
    public string? CreadoPor { get; set; }

    [StringLength(100)]
    public string? UltimaModificacionPor { get; set; }

    public DateTime? FechaModificacion { get; set; }

    public Kiosko? Kiosko { get; set; }
    public ICollection<ActivoFijoAuditoria> Auditorias { get; set; } = new List<ActivoFijoAuditoria>();
}

File: SistemaTicket/sistema_ticketero/lib/domain/entities/activo_fijo.dart
class ActivoFijo {
  final int id;
  final String tipoActivo;
  final String numeroActivo;
  final String? descripcion;
  final String? marca;
  final String? modelo;
  final String? serie;

  const ActivoFijo({
    this.id = 0,
    required this.tipoActivo,
    required this.numeroActivo,
    this.descripcion,
    this.marca,
    this.modelo,
    this.serie,
  });

  ActivoFijo copyWith({
    int? id,
    String? tipoActivo,
    String? numeroActivo,
    String? descripcion,
    String? marca,
    String? modelo,
    String? serie,
  }) {
    return ActivoFijo(
      id: id ?? this.id,
      tipoActivo: tipoActivo ?? this.tipoActivo,
      numeroActivo: numeroActivo ?? this.numeroActivo,
      descripcion: descripcion ?? this.descripcion,
      marca: marca ?? this.marca,
      modelo: modelo ?? this.modelo,
      serie: serie ?? this.serie,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'tipoActivo': tipoActivo,
        'numeroActivo': numeroActivo,
        'descripcion': descripcion,
        'marca': marca,
        'modelo': modelo,
        'serie': serie,
      };

  factory ActivoFijo.fromJson(Map<String, dynamic> json) => ActivoFijo(
        id: json['id'] as int? ?? 0,
        tipoActivo: (json['tipoActivo'] as String?) ?? '',
        numeroActivo: (json['numeroActivo'] as String?) ?? '',
        descripcion: json['descripcion'] as String?,
        marca: json['marca'] as String?,
        modelo: json['modelo'] as String?,
        serie: json['serie'] as String?,
      );
}
